pkg/pb/metainfo.proto

```diff
@@ -328,4 +340,139 @@ message SatStreamID {
     google.protobuf.Timestamp expiration_date = 6  [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
 
     bytes satellite_signature = 7;
+}
+
+//---------------------------
+// Segment
+//---------------------------
+
+message Segment {
+    bytes stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    SegmentPosition position = 2;
+
+    bytes encrypted_key_nonce = 3 [(gogoproto.customtype) = "Nonce", (gogoproto.nullable) = false];
+    bytes encrypted_key = 4;
+
+    int64 size_encrypted_data = 5; // refers to segment size not piece size
+
+    bytes encrypted_inline_data = 6;
+    repeated Piece pieces = 7;
+}
+
+message Piece {
+    int32 piece_num = 1;
+    bytes node = 2[(gogoproto.customtype) = "NodeID", (gogoproto.nullable) = false];
+}
+
+message SegmentPosition {
+    int32 part_number = 1;
+    int32 index = 2;
+}
+
+message SegmentBeginRequest {
+    bytes           stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    SegmentPosition position = 2;
+
+    int64 max_order_limit = 3;
+}
+
+message SegmentBeginResponse {
+    bytes    segment_id = 1 [(gogoproto.customtype) = "SegmentID", (gogoproto.nullable) = false];
+    repeated AddressedOrderLimit addressed_limits = 2;
+    bytes private_key = 3 [(gogoproto.customtype) = "PiecePrivateKey", (gogoproto.nullable) = false];
+}
+
+message SegmentCommitRequest {
+    bytes segment_id = 1 [(gogoproto.customtype) = "SegmentID", (gogoproto.nullable) = false];
+
+    bytes encrypted_key_nonce = 2 [(gogoproto.customtype) = "Nonce", (gogoproto.nullable) = false];
+    bytes encrypted_key = 3;
+
+    int64 size_encrypted_data = 4; // refers to segment size not piece size
+
+    repeated SegmentPieceUploadResult upload_result = 5;
+}
+
+message SegmentPieceUploadResult {
+    int32               piece_num = 1;
+    bytes               node_id = 2 [(gogoproto.customtype) = "NodeID", (gogoproto.nullable) = false];
+    orders.PieceHash    hash = 3;
+}
+
+// only for satellite use
+message SatSegmentID {
+    SatStreamID stream_id = 1;
+    int32    part_number = 2;
+    int32    index = 3;
```

Comment:
Does the part_number in message SatSegmentID and the part_number in message SegmentPosition refer to segment's position in the stream?

Reply:
Field `part_number` in general will be used for multipart upload in the future.